### PR TITLE
positional filter select box validation added and study browser radio butto…

### PR DESCRIPTION
**Pull request by kandaj**
…ns order changed

### CODE REVIEW

File: src/js/variant-widget/filters/eva-form-panel.js [hunk 1]
```diff
@@ -193,5 +193,31 @@ EvaFormPanel.prototype = {
...
 
+    },
+    validatePositionFilter : function(values) {
+        var validate = true;
+        _.each(values, function (region) {
+            if(region.match(/^[\w\d]+\:\d+\-\d+/)){
+                var start = region.split(':')[1].split('-')[0];
+                var end = region.split(':')[1].split('-')[1]
+                if (end - start > 1000000) {
+                    Ext.Msg.alert('Limit Exceeds', 'Please enter the region no more than 1000000 range');
```
Comment (cyenyxe) [inline src/js/variant-widget/filters/eva-form-panel.js]: Maybe something like "Please enter a region no larger than 1 million bases" could be a bit clearer?

File: src/js/variant-widget/filters/eva-form-panel.js [hunk 2]
```diff
@@ -193,5 +193,31 @@ EvaFormPanel.prototype = {
         });
         return form;
 
+    },
+    validatePositionFilter : function(values) {
+        var validate = true;
+        _.each(values, function (region) {
+            if(region.match(/^[\w\d]+\:\d+\-\d+/)){
```
Comment (cyenyxe) [inline src/js/variant-widget/filters/eva-form-panel.js]: I have checked online and this should accept chromosomes with underscores, but what about other symbols like dashes or dots?

Comment (kandaj) [inline src/js/variant-widget/filters/eva-form-panel.js]: do you what are the special characters other than [_.-] can a chromosome have?.

Comment (cyenyxe) [inline src/js/variant-widget/filters/eva-form-panel.js]: According to the VCF spec: 

> Valid contig names must follow the reference sequence names allowed by the SAM format ("[!-)+-<>-~][!-~]\*") excluding the characters "<>[]:*"

So you might be able to use that regular expression with the exclusion. Please implement some tests to check values are accepted or rejected accordingly.

Comment (kandaj) [inline src/js/variant-widget/filters/eva-form-panel.js]: I've updated the regex and web tests are added.

File: tests/mocha/variant_browser.js [hunk 3]
```diff
@@ -92,6 +98,30 @@ function variantSearchBySpeciesandChrLocation(driver){
...
     });
 }
+
+function positionFilterBoxValidation(driver){
+    driver.findElement(By.id("selectFilter-trigger-picker")).click();
+    driver.findElement(By.xpath("//li[text()='Chromosomal Location']")).click();
+    driver.findElement(By.id("speciesFilter-trigger-picker")).click();
+    driver.findElement(By.xpath("//li[text()='Mosquito / AaegL3']")).click();
+    driver.findElement(By.name("region")).clear();
+    driver.findElement(By.name("region")).sendKeys('supercont1.18:165624-165624');
```
Comment (cyenyxe) [inline tests/mocha/variant_browser.js]: Could some more tests like this be implemented? Some for chromosomes made only of letters, digits, alphanumeric, with different punctuation...

Comment (kandaj) [inline tests/mocha/variant_browser.js]: for
letter: X:10000000-11000000
digits: 2:4000000-4100000
alphanumeric: supercont1.18:165624-165624 are added.
do you know a example for chr with different punctuation?.

Comment (cyenyxe) [inline tests/mocha/variant_browser.js]: Not real cases, but you can create string like "contig,with,commas" for testing the regex is correct.

Comment (kandaj) [inline tests/mocha/variant_browser.js]: tests are updated.
1,13:12233-12234 this should open  alert box with "Please enter a valid region" msg.

**Pull request merged**